README.md

```diff
@@ -939,12 +940,69 @@ await pusher.UnsubscribeAllAsync().ConfigureAwait(false);
 Add this handler to recieve subscription count events. Read more about it [here](https://pusher.com/docs/channels/using_channels/events/#pushersubscription_count-1165820117)
 
 ```cs
+
 void PusherCountEventHandler(object sender, string data) {
     var dataAsObj = JsonConvert.DeserializeObject<SubscriptionCountData>(data);
     Console.WriteLine(dataAsObj.subscriptionCount);
 }
 
 pusher.CountHandler += PusherCountEventHandler;
+
+```
+
+## User authentication
+
+Allows you to send events to a user based on user id or terminating a user’s connections immediately. Read more about it [here](https://pusher.com/docs/channels/using_channels/user-authentication/)
+
+### Signing in
+
+The library provides an `HttpUserAuthenticator` implementation of `IUserAuthenticator` which makes an HTTP `POST` request to an authenticating endpoint. However, you can implement your own authentication mechanism if required.
+
+Setting up a authenticated user
+
+```cs
+
+// Create client
+Pusher pusher = new Pusher(Config.AppKey, new PusherOptions
+{
+    UserAuthenticator = new HttpUserAuthenticator("https:/some.authenticator.com/auth")
+    Cluster = Config.Cluster,
+});
+pusher.Error += ErrorHandler;
+
+// Sign in
+pusher.User.Signin();
+
+// Connect
+await pusher.ConnectAsync().ConfigureAwait(false);
+await pusher.User.SigninDoneAsync();
+
+```
+
+### HttpUserAuthenticator
+
+The implementation of the `HttpUserAuthenticator` class which provides the default implementation of an 

```

Comment:
I am not sure this sentence makes sense. 
It doesn't read well, it is difficult to know what is default and what has been modified. Could we re-write?